### PR TITLE
[MilkmApp] Add Curve Price Checker on gnosis

### PR DESCRIPTION
**Pull request by yvesfracari**

### CODE REVIEW

File: apps/cow-tools/src/app/milkman/(components)/SingleOrderForm/PriceChecker.tsx [hunk 1]
```diff
@@ -209,9 +214,9 @@ function ArrayPriceCheckerInput({
                   return (
                     <Table.BodyCell classNames="align-top" key={arg.name}>
                       {arg.type.includes("bool") ? (
-                        <div className="flex items-center justify-center gap-x-2">
+                        <div className="flex items-center justify-center gap-x-2 mt-2">
                           <input
-                            className="h-5 w-5"
+                            className="h-5 w-5 mt-2"
```
Comment (yvesfracari) [inline apps/cow-tools/src/app/milkman/(components)/SingleOrderForm/PriceChecker.tsx:219]: minor style fix -- align checkbox with other inputs

File: apps/cow-tools/src/app/milkman/(components)/OrderTable/TransactionInfo.tsx [hunk 2]
```diff
@@ -17,7 +18,7 @@ export function TransactionInfo({
   order: AllTransactionFromUserQuery["users"][0]["transactions"][0]["swaps"][0];
 }) {
   const priceChecker = getPriceCheckerFromAddressAndChain(
-    order.chainId as 5,
+    order.chainId as ChainId,
```
Comment (luizakp) [inline apps/cow-tools/src/app/milkman/(components)/OrderTable/TransactionInfo.tsx:21]: 👍

File: apps/cow-tools/src/app/milkman/page.tsx [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+import { gnosis, goerli } from "viem/chains";
 
 export default function Page() {
-  //TODO: change redirect to default network BLEU-366
-  redirect("/milkman/goerli");
+  const {
+    safe: { chainId },
+  } = useSafeAppsSDK();
+
+  if (chainId == goerli.id || chainId == gnosis.id) {
```
Comment (luizakp) [inline apps/cow-tools/src/app/milkman/page.tsx:13]: later on you can create a function to check if the chain ID is supported, we have something like this on balancer-tools

File: apps/cow-tools/src/lib/schema.ts [hunk 4]
```diff
@@ -141,8 +142,8 @@ export const generatePriceCheckerSchema = ({
             functionName: "checkPrice",
             args: [
               1, // we're just interested in call revert or not, so this value is not important
-              tokenBuyAddress,
               tokenSellAddress,
+              tokenBuyAddress,
```
Comment (luizakp) [inline apps/cow-tools/src/lib/schema.ts:146]: the order was swapped before?

**Comment by luizakp** [comment]
Just something to check later, when I don't have transaction it keeps loading forever
image

**Comment by luizakp** [comment]
Curve had no parameters, no? I can't get pass that
image

**Comment by yvesfracari** [comment]
What tokens did you use to test? If there is no route for the pair on curve it will return an error

**Review by luizakp — APPROVED** [review]
tried with WXDAI for USDC and it worked now

**Pull request merged**